internal/provider/telemetry_resource.go

```diff
@@ -204,18 +226,17 @@ func (r *TelemetryResource) ImportState(ctx context.Context, req resource.Import
 }
 
 // withModuleSourceAndVersion updates the module source and version based on the module path.
-func withModuleSourceAndVersion(data TelemetryResourceModel) TelemetryResourceModel {
-	data.ModuleSource = basetypes.NewStringNull()
-	data.ModuleVersion = basetypes.NewStringNull()
-	if data.ModulePath.IsNull() || data.ModulePath.IsUnknown() {
-		return data
-	}
-	module, err := parseModulesJson(data.ModulePath.ValueString())
-	if err != nil {
-		return data
+func withModuleSourceAndVersion[T moduleSource](data T) T {
```

Comment:
Could we move this func to the interface file, as it no longer only pertains to the telemetry resource?